packages/next/src/build/index.ts

```diff
@@ -2129,17 +2129,21 @@ export default async function build(
                           const appConfig = workerResult.appConfig || {}
                           if (appConfig.revalidate !== 0) {
                             const isDynamic = isDynamicRoute(page)
+                            const prerenderedRoutes =
+                              workerResult.prerenderedRoutes
+                            const generateStaticParamsIsDefined =
+                              prerenderedRoutes !== undefined
                             const hasGenerateStaticParams =
-                              workerResult.prerenderedRoutes &&
-                              workerResult.prerenderedRoutes.length > 0
+                              generateStaticParamsIsDefined &&
+                              prerenderedRoutes.length > 0
```

Comment:
Storing `prerenderedRoutes` in a variable allows TypeScript to be confident that `prerenderedRoutes` is truthy and won't change between the definition of the two new variables here.

This latest rebase gives `hasGenerateStaticParams` its original name again to prioritize compatibility.